### DISCUSSION TITLE
Best way of sharing a mergerfs pool to multiple VMs

### DISCUSSION THREAD

**Discussion by BeyondEvil** [trapexit/mergerfs#1349] [Q&A] [answered]
Given [this statement](https://github.com/trapexit/mergerfs/discussions/1304#discussioncomment-8475112), is there a correct/better way of sharing a mergerfs pool to multiple VMs than using NFS?

I'm running Proxmox on a single machine/node and I'm planning on running a Kubernetes cluster on it. So I need a way of making the mergerfs pool available to pods (as PV).

Note that the VM storage itself is coming from a ZFS mirror on entirely different drives.

**Comment by trapexit** [comment]
There is no "correct". Sharing filesystems (nothing to do with mergerfs) across numerous systems requires a remote filesystem. There are really only two remote filesystems built for such things: Samba/CIFS/SMB and NFS.

**Reply by BeyondEvil** [reply]
Thanks for the quick response!

And are Samba/CIFS/SMB and NFS equally good/bad w.r.t. mergerfs - or is it inconsequential?

Also, I'm having trouble understanding what "out of band" means in this context. Does it mean making changes to files/directories outside of NFS?

If I don't have any out of band changes, will NFS work for my use case?

**Reply by trapexit** [reply, marked as answer]
https://github.com/trapexit/mergerfs?tab=readme-ov-file#remote-filesystems

Did you read the docs on the topic?

**Reply by trapexit** [reply]
Out of band means... out of band. NFS and SMB share some filesystem /mnt/foo and in-band means you use NFS or SMB. Out of band means you don't. You access the filesystem out of band of the share. IE directly.

**Reply by BeyondEvil** [reply]
Sorry, non-native english speaker, and have never come across the term "in/out of band" previously.

But it seems my understanding is correct nevertheless.

Apologies, I had not found my way to the specific docs - I had found the statement when I was searching for a different issue.

I'll see what the support (and possible downsides) is for my use case with SMB/Samba.

Thanks again!
